internal/manifests/collector/daemonset.go

```diff
@@ -19,44 +19,46 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+	"github.com/open-telemetry/opentelemetry-operator/internal/api/convert"
 	"github.com/open-telemetry/opentelemetry-operator/internal/manifests"
 	"github.com/open-telemetry/opentelemetry-operator/internal/manifests/manifestutils"
 	"github.com/open-telemetry/opentelemetry-operator/internal/naming"
 )
 
 // DaemonSet builds the deployment for the given instance.
 func DaemonSet(params manifests.Params) *appsv1.DaemonSet {
```

Comment:
I wonder if we should just have the params here work off of the v1alpha2 type rather than converting it here.

Reply:
I agree, but that change had other implications for example on `opampbridge` and `targetallocator`.

Reply:
good question, I assume it tests and all the things need to be changed in at least the other sub packages.